account_invoice_open_sale/models/account_invoice.py

```diff
@@ -23,12 +23,10 @@ def _compute_has_sale_order(self):
         JOIN sale_order_line_invoice_rel AS rel ON ail.id = rel.invoice_line_id
         WHERE ail.invoice_id IN %s
         """, (tuple(self.ids),))
-        invoice_ids = self.env.cr.fetchall()
 
-        if not invoice_ids:
-            return True
+        invoice_ids = [invoice_id[0] for invoice_id, in self.env.cr.fetchall()]
```

Comment:
Does this work? Looks to me you should either do
```
[row[0] for row in self.env.cr.fetchall()]
```
Or
```
[invoice_id for invoice_id, in self.env.cr.fetchall()]
```
Writing `invoice_id,` is a shorthand to unpack a tuple with only one member.